pkg/tracking/store/sql/experiments.go

```diff
@@ -252,3 +253,63 @@ func (s TrackingSQLStore) GetExperimentByName(
 
 	return experiment.ToEntity(), nil
 }
+
+func (s TrackingSQLStore) SetExperimentTag(
+	ctx context.Context, experimentID, key, value string,
+) *contract.Error {
+	experiment, err := s.GetExperiment(ctx, experimentID)
+	if err != nil {
+		return err
+	}
+
+	// check experiment is active
+	if models.LifecycleStage(experiment.LifecycleStage) != models.LifecycleStageActive {
+		return contract.NewError(
+			protos.ErrorCode_INVALID_PARAMETER_VALUE,
+			fmt.Sprintf(
+				"The experiment %q must be in the 'active' state.\n"+
+					"Current state is %q.",
+				experiment.ExperimentID,
+				experiment.LifecycleStage,
+			),
+		)
+	}
+
+	idInt, convertErr := strconv.ParseInt(experimentID, 10, 32)
```

Comment:
We have this as well in `GetExperiment`, please also extract and re-use this logic.